src/memray/reporters/flamegraph.py

```diff
@@ -117,87 +261,78 @@ def _from_any_snapshot(
                 size = record.size
                 n_allocations = record.n_allocations
 
-            if size is not None:
-                root["value"] += size
-            if n_allocations is not None:
-                root["n_allocations"] += n_allocations
-
-            current_frame_id = 0
-            current_frame = root
+            record_data: RecordData = {
+                "thread_name": thread_id,
+                "size": size,
+                "n_allocations": n_allocations,
+                "intervals": intervals,
+            }
```

Comment:
I'm gonna move this to be defined in each branch of the `if` above, so that we don't need `intervals` and `size` and `n_allocations` and `thread_id` as local variables in addition to values in this dict, for the same reason as above - limiting the number of local variables here will help future readers.